### ISSUE TITLE
Emacs 25 changes to system-name break smtp.el

### ISSUE THREAD

**Issue by jech** [wanderlust/wanderlust#96]
In Emacs 24, both the variable and the function `system-name` return an FQDN.  In Emacs 25, this is no longer the case — both return an unqualified hostname.  This breaks `smtp-make-fqdn`, so that sending mail over SMTP says

    Cannot generate valid FQDN

While this is easily solved by setting `smtp-fqdn` or `smtp-local-domain`, there are two issues here:

  * we should really try to find out the FQDN automatically, even under Emacs 25; and
  * the error message should be more informative (point at `smtp-fqdn` and `smtp-local-domain`).

**Comment by ikazuhiro** [comment]
>  we should really try to find out the FQDN automatically, even under Emacs 25

If there is a easy way to get FQDN from Emacs lisp, I would try to do so.  But because FQDN is not available for all machines, I think the featue is not essential.

> the error message should be more informative (point at `smtp-fqdn` and `smtp-local-domain`).

Done at https://github.com/wanderlust/flim/commit/a3baf0f785828e21ce9df349dec0b0d958777e11.

**Comment by jech** [comment]
Right.  http://emacs.stackexchange.com/questions/12649/local-hosts-fqdn-in-emacs-25

**Comment by jech** [comment]
What about this instead?

    diff -u smtp.el~ smtp.el
    --- smtp.el~	2015-05-19 21:53:03.998786545 +0200
    +++ smtp.el	2015-05-24 00:29:58.733820265 +0200
    @@ -249,7 +249,7 @@
            ((string-match "[^.]\\.[^.]" system-name)
            system-name)
            (t
    -	(error "Cannot generate valid FQDN"))))))
    +        (concat system-name ".localdomain"))))))

     (defun smtp-find-connection (buffer)
       "Find the connection delivering to BUFFER."

Granted, it violates RFC 5321 Section 4.1.4, but it will work if the user is using a smarthost — which he probably is, since `smtp-send-by-myself` is not likely to work.

**Comment by anarcat** [comment]
i just used `(car (split-string (shell-command-to-string "hostname -f")))` here - even though I don't actually use Wanderlust. :)
